### PR TITLE
Pytorch aurora

### PR DESCRIPTION
**Pull request by FilippoSimini**

### CODE REVIEW

**Comment by FilippoSimini** [comment]
it's complaining about a wrongly formatted link:
```
WARNING -  Doc file 'aurora/data-science/frameworks/pytorch.md' contains a link '#example%3A-training-a-pytorch-model-on-a-single-gpu-tile', but there is no such anchor on this page.
```
the issue is probably with the colon `:` after `example`

File: docs/aurora/data-science/frameworks/pytorch.md [hunk 1]
```diff
@@ -38,77 +36,194 @@ print(f'Device properties = {torch.xpu.get_device_properties()}')
...
+    	_XpuDeviceProperties(name='Intel(R) Data Center GPU Max 1550', platform_name='Intel(R) Level-Zero', type='gpu', driver_version='1.3.30872', total_memory=131072MB, max_compute_units=896, gpu_eu_count=896, gpu_subslice_count=112, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1)
+    	```
+    	
+    More information and details are available through the [Level Zero Specification Documentation - Affinity Mask](https://oneapi-src.github.io/level-zero-spec/level-zero/latest/core/PROG.html?highlight=affinity#affinity-mask)
+
+
+
+## Code changes to run PyTorch on Aurora GPUs
+
+[Intel extension for PyTorch (IPEX)](https://pytorch.org/tutorials/recipes/recipes/intel_extension_for_pytorch.html) is an [open-source project](https://github.com/intel/intel-extension-for-pytorch) that extends PyTorch with optimizations for extra performance boost on Intel hardware, including Intel GPUs.
```
Comment (felker) [inline docs/aurora/data-science/frameworks/pytorch.md:98]:
```suggestion
[Intel Extension for PyTorch (IPEX)](https://pytorch.org/tutorials/recipes/recipes/intel_extension_for_pytorch.html) is an [open-source project](https://github.com/intel/intel-extension-for-pytorch) that extends PyTorch with optimizations for extra performance boost on Intel hardware, including Intel GPUs. 
```

might want to rephrase the last part--- maybe to:
```suggestion
[Intel Extension for PyTorch (IPEX)](https://pytorch.org/tutorials/recipes/recipes/intel_extension_for_pytorch.html) is an [open-source project](https://github.com/intel/intel-extension-for-pytorch) that extends PyTorch with optimizations for extra performance boost on Intel CPUs and enables the use of Intel GPUs.
```

there is still no `xpu` functionality in upstream PyTorch yet, right?

File: docs/aurora/data-science/frameworks/pytorch.md [hunk 2]
```diff
@@ -38,77 +36,194 @@ print(f'Device properties = {torch.xpu.get_device_properties()}')
...
-```
-With this environmental variable, we expect the output to be `6` -- the number
-of GPUs available on an Aurora compute node. All the `API` calls involving 
-`torch.cuda`, should be replaced with `torch.xpu`, as shown in the above 
-example.
+    ``` { .bash .no-copy }
+        GPU availability: True
+        Number of tiles = 12
+        Current tile = 0
+        Curent device ID = <intel_extension_for_pytorch.xpu.device object at 0x1540a9f25790>
```
Comment (felker) [inline docs/aurora/data-science/frameworks/pytorch.md:45]:
```suggestion
        Current device ID = <intel_extension_for_pytorch.xpu.device object at 0x1540a9f25790>
```

also see L34

File: docs/aurora/data-science/frameworks/pytorch.md [hunk 3]
```diff
@@ -122,258 +237,176 @@ PyTorch, users generally need to manage casting and loss scaling manually,
 though context managers and function decorators can provide easy tools to do 
 this.
 
-2. PyTorch has a `JIT` module as well as backends to support op fusion, similar 
-to TensorFlow's `tf.function` tools. 
-Please see [TorchScript](https://pytorch.org/docs/stable/jit.html) for more 
-information.
+2. PyTorch has a `JIT` module as well as backends to support op fusion, similar to TensorFlow's `tf.function` tools. See [TorchScript](https://pytorch.org/docs/stable/jit.html) for more information.
```
Comment (felker) [inline docs/aurora/data-science/frameworks/pytorch.md:240]: actually above, a typo:

> In general, the way to do this is via the 
PyTorch Automatic Mixed Precision package (AMP), as **descibed** in the

File: docs/aurora/data-science/frameworks/tensorflow.md [hunk 4]
```diff
@@ -204,7 +204,7 @@ Some Aurora specific details might be helpful to you.
 The following environmental variables should be set on the batch submission 
 script (PBSPro script) in the case of attempting to run beyond 16 nodes.
 
---8<-- "./docs/aurora/data-science/frameworks/pytorch.md:commononecclenv"
+--8<-- "./docs/aurora/data-science/frameworks/oneCCL.md:onecclenv"
```
Comment (felker) [inline docs/aurora/data-science/frameworks/tensorflow.md:207]: typos in this file, courtesy of copilot:

Line 6: "trouble shooting" should be "troubleshooting".
Line 60: "Intel extension for TensorFLow is has" should be "Intel extension for TensorFlow has".
Line 101: "descibed" should be "described".
Line 219: "use" should be "used".
Line 243: "to changed" should be "to change".
